feat(Stripe Node): Add meter events for billing (deprecated)

jason-schell opened the pull request:
## Summary

Add the meter event to the Stripe node, so that it's possible to use the [Create a billing meter event](https://docs.stripe.com/api/billing/meter-event/create)-API (v1).


## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [x] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

jason-schell committed:
Stripe node: Added meter events for billing.

jason-schell committed:
Added unit tests

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=18751) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=18751) before we can accept your contribution.<br/><hr/>**Jason Schell** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=18751) it.</sub>

jason-schell committed:
Remove type assertions, use type annotations.

jason-schell committed:
- Replace type assertions with typed generics for better type safety
- Add proper type guards for nested property access
- Update value description to accurately reflect negative values support
- Improve error message specificity for invalid value types
- Remove redundant type check with proper generic typing
- Add test cases for edge cases and error conditions
- Expand schema to include optional Stripe response fields

jason-schell committed:
Bugfix: Timestamp Conversion changed from string to integer

jason-schell committed:
Improving quality at custom payload processing and adding more tests for custom payload fields processing, errors, edge cases reg. timestamp conversion and negative value testing.
